app/routes/__index/logout.tsx

```diff
@@ -0,0 +1,64 @@
+import { redirect } from '@remix-run/node';
+import { Form, useLoaderData } from '@remix-run/react';
+import { AbsoluteCenter, Button, Container, Stack, Text } from '@chakra-ui/react';
+import { createLogoutRequest } from '~/lib/saml.server';
+import { getUsername, logout } from '~/session.server';
+
+import type { ActionArgs, LoaderArgs } from '@remix-run/node';
+
+declare global {
+  var __tempUser__: string | undefined;
```

Comment:
This isn't going to work.  If you have 2 users doing this at the same time, only 1 will be able to store their state here.  You need to store this info in the browser or cookie or in some DOM state or something.